src/model/ModelResources.i

```diff
@@ -174,6 +174,8 @@ MODELOBJECT_TEMPLATES(CoilCoolingDXCurveFitPerformance);
 MODELOBJECT_TEMPLATES(CoilCoolingDXCurveFitOperatingMode);
 MODELOBJECT_TEMPLATES(CoilCoolingDXCurveFitSpeed);
 
+MODELOBJECT_TEMPLATES(HeatExchangerDesiccantBalancedFlowPerformanceDataType1);
```

Comment:
C# SWIG: `HeatExchangerDesiccantBalancedFlowPerformanceDataType1` is swigg'ed in ModelResources.i. 
`HeatExchangerDesiccantBalancedFlow` is swigg'ed **later** in ModelHVAC.i

For reference, the order is defined here: https://github.com/NREL/OpenStudio/blob/28c3fc19e4d0f0a8c2168a6242f3dbf3dc32dcde/src/model/CMakeLists.txt#L2255-L2271

So that means that the `std::vector<HeatExchangerDesiccantBalancedFlow> HeatExchangerDesiccantBalancedFlowPerformanceDataType1::heatExchangerDesiccantBalancedFlows() const;` must be ignored at this point, and reimplemented using Partial classes in ModelHVAC.i or C# won't know the type...

The broader question is whether it was the right choice to have made
* `HeatExchangerDesiccantBalancedFlowPerformanceDataType1` a `ResourceObject`, **and**
* to have it swigg'ed in ModelResources.i versus in ModelHVAC.i

One thing I cannot answer right away, but I can't help notice a pattern with other HVACComponents like `CoilCoolingDXVariableSpeed` and its `CoilCoolingDXVariableSpeedSpeedData` which is a ParentObject (because it has curves probably) but still is defined in the ModelHVAC.i.

After some consideration, I think it's a valid design decision to have it like that. I think it should be a ResourceObject indeed, and given that, it makes sense to have in ModelResources.i. @kbenne might have a different opinion though?  I'll do the partial classes in the meantime, it's low effort and can be reverted.

cc @joseph-robertson

Reply:
cf 2f79955ec6b4228e8b8c9a4276333fa069903485